extensions/arc/notebooks/arc/content/postgres/tsg100-troubleshoot-postgres.ipynb

```diff
@@ -0,0 +1,590 @@
+    "\n",
+    "            kubectl get nodes -o jsonpath={.items[?(@.metadata.annotations.pv-candidate=='data-pool')].metadata.name}\n",
+    "\n",
+    "        Need to actually pass in as '\"':\n",
+    "\n",
+    "            kubectl get nodes -o jsonpath={.items[?(@.metadata.annotations.pv-candidate=='\"'data-pool'\"')].metadata.name}\n",
+    "\n",
+    "        The ' quote approach, although correct when pasting into Windows cmd, will hang at the line:\n",
+    "        \n",
+    "            `iter(p.stdout.readline, b'')`\n",
+    "\n",
+    "        The shlex.split call does the right thing for each platform, just use the '\"' pattern for a '\n",
+    "    \"\"\"\n",
+    "    MAX_RETRIES = 5\n",
+    "    output = \"\"\n",
+    "    retry = False\n",
+    "\n",
+    "    global first_run\n",
+    "    global rules\n",
+    "\n",
+    "    if first_run:\n",
+    "        first_run = False\n",
+    "        rules = load_rules()\n",
+    "\n",
+    "    # When running `azdata sql query` on Windows, replace any \\n in \"\"\" strings, with \" \", otherwise we see:\n",
+    "    #\n",
+    "    #    ('HY090', '[HY090] [Microsoft][ODBC Driver Manager] Invalid string or buffer length (0) (SQLExecDirectW)')\n",
+    "    #\n",
+    "    if platform.system() == \"Windows\" and cmd.startswith(\"azdata sql query\"):\n",
+    "        cmd = cmd.replace(\"\\n\", \" \")\n",
+    "\n",
+    "    # shlex.split is required on bash and for Windows paths with spaces\n",
+    "    #\n",
+    "    cmd_actual = shlex.split(cmd)\n",
+    "\n",
+    "    # Store this (i.e. kubectl, python etc.) to support binary context aware error_hints and retries\n",
+    "    #\n",
+    "    user_provided_exe_name = cmd_actual[0].lower()\n",
+    "\n",
+    "    # When running python, use the python in the ADS sandbox ({sys.executable})\n",
+    "    #\n",
+    "    if cmd.startswith(\"python \"):\n",
+    "        cmd_actual[0] = cmd_actual[0].replace(\"python\", sys.executable)\n",
+    "\n",
+    "        # On Mac, when ADS is not launched from terminal, LC_ALL may not be set, which causes pip installs to fail\n",
+    "        # with:\n",
+    "        #\n",
+    "        #    UnicodeDecodeError: 'ascii' codec can't decode byte 0xc5 in position 4969: ordinal not in range(128)\n",
+    "        #\n",
+    "        # Setting it to a default value of \"en_US.UTF-8\" enables pip install to complete\n",
+    "        #\n",
+    "        if platform.system() == \"Darwin\" and \"LC_ALL\" not in os.environ:\n",
+    "            os.environ[\"LC_ALL\"] = \"en_US.UTF-8\"\n",
+    "\n",
+    "    # When running `kubectl`, if AZDATA_OPENSHIFT is set, use `oc`\n",
+    "    #\n",
+    "    if cmd.startswith(\"kubectl \") and \"AZDATA_OPENSHIFT\" in os.environ:\n",
+    "        cmd_actual[0] = cmd_actual[0].replace(\"kubectl\", \"oc\")\n",
+    "\n",
+    "    # To aid supportabilty, determine which binary file will actually be executed on the machine\n",
```

Comment:
supportability

Reply:
Thanks, good find.  This run function is referenced from an existing snippet that I didn't author, but I'll send a PR to aris fixing it.